src/components/navigation/nav-tab-buttons.tsx

```diff
@@ -30,14 +30,18 @@ export class NavTabButtons extends BaseComponent<IProps, IState> {
 
   public render() {
     const { tabs } = this.props;
-    const { ui } = this.stores;
+    const { ui, user, supports } = this.stores;
     return (
       <div className={`nav-tab-buttons ${ui.navTabContentShown ? "hidden" : ""}`}>
         { tabs?.map((tabSpec, i) => {
             const tabClass = `nav-tab tab-${tabSpec.tab}`;
             return (
               <div key={tabSpec.tab} className={tabClass} onClick={this.handleTabButtonClick(tabSpec.tab)}>
                 {tabSpec.label}
+                {(tabSpec.tab === "supports"
+                  && user.isStudent
+                  && supports.hasNewTeacherSupports(user.lastSupportViewTimestamp))
+                  && <div className={`support-badge`}></div>}
```

Comment:
Since this code occurs multiple times, I'd make a `<SupportBadge>` component which could be called like this:
```typescript
  <SupportBadge user={user} supports={supports} />
```